src/ray/core_worker/core_worker.cc

```diff
@@ -716,13 +717,16 @@ Status CoreWorker::SerializeActorHandle(const ActorID &actor_id,
 }
 
 bool CoreWorker::AddActorHandle(std::unique_ptr<ActorHandle> actor_handle) {
+  std::lock_guard<std::mutex> lock(actor_handles_mutex_);
```

Comment:
Why was this not necessary before?

Reply:
Previously we don't store `ActorHandle` instances in a map, the instances are hold in user code. Due to multi-threading, now we need to protect the actor handles map.